command/command.go

```diff
@@ -203,3 +198,48 @@ func (s *shiftUpCommand) Execute() {
 func (s *shiftDownCommand) Execute() {
 	s.ShiftDown()
 }
+
+// Manages interactions between Player and Transmission
+type shiftController struct {
+	*transmission
+	*shiftDownCommand
+	*shiftUpCommand
+}
+
+// Returns a new shiftController with pointers to a transmission, shiftUpCommand and shiftDownCommand.
+// Things to consider: once we have multiple players each shifting their own transmission, should those interactions all be handled by a single shiftController, or should each player/transmission pair get it's own shiftController?
+func newShiftController() *shiftController {
+	t := newTransmission()
+	d := newShiftDownCommand(t)
+	u := newShiftUpCommand(t)
+	return &shiftController{t, d, u}
+}
+
+// Handles shifting a transmission up for a player
+func (s *shiftController) ShiftUpAction(c *CommandsList) {
+	s.shiftUpCommand.Execute()
+
+	// don't know if this is the best approach, but I don't think the alternative - returning the shiftCommand as a "success" value and having the Player's shiftUp method push that onto the slice - is really any better
```

Comment:
An alternative would be to give the shift controller its own CommandsList. That would be fine if you only want to let the shifter control the shift state and you don't want global access to it. Otherwise, I would implement a CommandsList interface with all the same methods the current struct has now, and I would change CommandsList struct to DefaultCommandsList - this would allow you to implement your own list functions by hooking into the default. For example

    type shifterCommandsList struct {
        DefaultCommandsList
    }
   
    func (s *shifterCommandsList) Push(c Command) {
         someGlobalList.Push(c)
         s.DefaultCommandsList.Push(c)
    }

This way it gets added to both lists but you still have access to both.